src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -8,64 +8,73 @@
+    private static final char FALSE = 'F';
+
+    /**State before device is created, spun, poked, or peeked. */
     private static final int STATE_NOT_CREATED = 0;
 
-    /**State after device is created */
+    /**State after device is created. */
     private static final int STATE_CREATED = 1;
 
     /**State after device is spun.*/
     private static final int STATE_SPUN = 2;
 
-    /**State after device is peeked */
+    /**State after device is peeked. */
     private static final int STATE_PEEKED = 3;
 
-    /**State after device is poked */
+    /**State after device is poked. */
     private static final int STATE_POKED = 4;
 
     /**Holds the state of the unlock.*/
     private static int state = STATE_NOT_CREATED;
 
     /**
-     * Static device to unlock
+     * Static device to unlock.
      */
     private static Device dev;
 
     /** Pattern requested from doPeek. */
     private static CharSequence peekedPattern = null;
 
-    /** Char we are changing device bits to default to 'T'*/
-    private static char changeBitTo = 'T';
+    /** Char we are changing device bits to default to 'T'.*/
+    private static char changeBitTo = TRUE;
 
-    /** Number of Bits for Device */
-    private static final int numOfBits = 4;
+    /** Number of Bits for Device. */
+    private static final int NUM_OF_BITS = 4;
 
-    /** Number of bits that are disclosed*/
-    private static final int numOfBitsDisclosed = 2;
+    /** Number of bits that are disclosed.*/
+    private static final int NUM_OF_BITS_DISCLOSED = 2;
 
     /** Log of all SPIN/PEEK/POKE actions performed. */
     private static StringBuilder traceLog = new StringBuilder();
 
     /**
-     * Unlocks a device-controlled resource.
-     * This method must be guaranteed to halt, regardless of
-     * whether or not it successfully unlocked the resource.
-     * @param dev the device controlling the resource to unlock
-     * @return true if the resource is unlocked (all bits in the
-     *         device are now identical); false otherwise
+     * Unlocks a resource controlled by a 4-bit/2-disclosed device. Behavior is unspecified if parameter is not a reference to a valid 4-bit/2-disclosure device.
+     * @param dev the device controlling the resource to unlock; must be a 4-bit device with 2 peek/poke bits.
+     * @return true if the resource is unlocked (all bits in the device are now identical); false otherwise
      */
     public static boolean unlock(final Device dev) {
-        if (dev != null) {
+        if (dev == null) {
+            return false;
+        } else {
             state = STATE_CREATED;
         }
+        clearTrace();
         FourBitTwoDisclosureDeviceUnlocker.dev = dev;
         boolean isUnlocked = doSpin();
         List<CharSequence> perms = getPermutations();
-        int n = numOfBits;
-        while ((!isUnlocked) && (n > 0)) {
+        int n = 1;
+        while ((!isUnlocked) && (n <= NUM_OF_BITS)) {
```

Comment:
Here is where we implement the change.

It now goes from 1 to NUM_OF_BITS instead of NUM_OF_BITS to 1.